Lib/threading.py

```diff
@@ -1144,6 +1160,21 @@ def join(self, timeout=None):
             # historically .join(timeout=x) for x<0 has acted as if timeout=0
             self._wait_for_tstate_lock(timeout=max(timeout, 0))
 
+        if self._is_stopped:
+            self._join_os_thread()
```

Comment:
Just to be sure I've understood, the only time `self._is_stopped` would be False is if we hit the timeout.  Wouldn't that have raised an exception?

Is the condition here just as a safety precaution?  I suppose an assert would do:

```suggestion
        assert self._is_stopped
        self._join_os_thread()
```

Reply:
You're right, I'll make the change.